ts/adaptors/lite/Parser.ts

```diff
@@ -345,18 +345,19 @@ export class LiteParser implements MinDOMParser<LiteDocument> {
   /**
    * @param {LiteAdaptor} adaptor  The adaptor for managing nodes
    * @param {LiteElement} node     The node to serialize
+   * @param {boolean} xml          True when producing XML, false for HTML
    * @return {string}              The serialized element (like outerHTML)
    */
-  public serialize(adaptor: LiteAdaptor, node: LiteElement): string {
+  public serialize(adaptor: LiteAdaptor, node: LiteElement, xml: boolean = false): string {
     const SELF_CLOSING = (this.constructor as typeof LiteParser).SELF_CLOSING;
     const CDATA = (this.constructor as typeof LiteParser).CDATA_ATTR;
     const tag = adaptor.kind(node);
     const attributes = adaptor.allAttributes(node).map(
       (x: AttributeData) => x.name + '="' + (CDATA[x.name] ? x.value : this.protectAttribute(x.value)) + '"'
     ).join(' ');
     const html =
-      '<' + tag + (attributes ? ' ' + attributes : '') + '>'
-      + (SELF_CLOSING[tag] ? '' : adaptor.innerHTML(node) + '</' + tag + '>');
+      '<' + tag + (attributes ? ' ' + attributes : '')
+      + (SELF_CLOSING[tag] ? (xml ? ' />' : '>') : '>' + adaptor.innerHTML(node) + '</' + tag + '>');
```

Comment:
In XML any self-closing tag will close, regardless of whether there is `innerHTML` or not. Is that really intended?
In other words: `self_closing` is not the same as an empty tag?

Reply:
The purpose, here, is to get HTML to serialize as XML.  I'm not trying to handle all empty tags as self closing, as `<tag></tag>` is valid in that case.  This is to handle HTML where something like `<input...>` doesn't have a close tag or a `/>`, but needs to in order to be used in `<annotation-xml>` (unclosed tags cause parsing errors otherwise).  The `SELF_CLOSING` array lists those, and `innerHTML` uses that to generates HTML that doesn't have a close tag when it is not needed in HTML (like for `<input>`).  We don't want to do that for XML output.  Everything else gets a close tag already, so that is why `SELF_CLOSING` is the trigger for `/>` in this case.  Again, it is not about empty content (though it could be extended to do that, but that wasn't the intent).  

I do see, however, that `innerHTML` isn't the right thing to use, here, as the content needs to be serialized as XML as well.   So I will have to fix that.